libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java

```diff
@@ -534,44 +486,153 @@ public static String createResizedImageWithMaxWidth(Context context, String path
+            // Image width is within limits; don't resize
+            return path;
+        }
+
+        Uri imageUri = Uri.parse(path);
+        if (context == null || imageUri == null) {
+            return path;
+        }
+
+        File resizedImageFile;
+        FileOutputStream out;
+
+        try {
+            resizedImageFile = File.createTempFile("wp-image-", fileExtension);
+            out = new FileOutputStream(resizedImageFile);
+        } catch (IOException e) {
+            AppLog.e(AppLog.T.POSTS, "Failed to create the temp file on storage. Use the full picture instead.");
+            return path;
+        } catch (SecurityException e) {
+            AppLog.e(AppLog.T.POSTS, "Can't write the tmp file due to security restrictions. Use the full picture instead.");
+            return path;
+        }
+
+        try {
+            boolean res = resizeImageAndWriteToStream(context, imageUri, fileExtension, maxWidth, orientation, 90, out);
+            if (!res) {
+                AppLog.w(AppLog.T.POSTS, "Failed to compress the resized image. Use the full picture instead.");
+                return path;
+            }
+        } catch (IOException e) {
+            AppLog.e(AppLog.T.POSTS, "Failed to create resized image. Use the full picture instead.");
+            return path;
+        } catch (OutOfMemoryError e) {
+            AppLog.e(AppLog.T.POSTS, "Can't resize the picture due to low memory. Use the full picture instead.");
+            return path;
+        } finally {
+            // close the stream
+            try {
+                out.flush();
+                out.close();
+            } catch (IOException e) {
+            }
+        }
+
+        String tempFilePath = resizedImageFile.getPath();
+        if (!TextUtils.isEmpty(tempFilePath)) {
+            return tempFilePath;
+        } else {
+            AppLog.e(AppLog.T.POSTS, "Failed to create resized image. Use the full picture instead.");
+        }
+
+        return path;
+    }
+
+
+    /**
+     * nbradbury - 21-Feb-2014 - similar to createThumbnail but more efficient since it doesn't
+     * require passing the full-size image as an array of bytes[]
+     */
+    public static byte[] createThumbnailFromUri(Context context,
+                                                Uri imageUri,
+                                                int maxWidth,
+                                                String fileExtension,
+                                                int orientation) {
+        if (context == null || imageUri == null || maxWidth <= 0)
+            return null;
+
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        try {
+            boolean res = resizeImageAndWriteToStream(context, imageUri, fileExtension, maxWidth, orientation, 90, stream);
+            if (!res) {
+                AppLog.w(AppLog.T.POSTS, "Failed to compress the resized image. Use the full picture instead.");
```

Comment:
Checked and this could be eventually end up being used in WPHtml, which is used in `CommentAdapter` here https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/comments/CommentAdapter.java#L440
So let's better parameterize the log tag